### PR TITLE
Describe obsoletion of app.config sections

### PR DESCRIPTION
**Pull request by WilliamBZA**
Warns about the use of `app.config`, `IProvideConfiguration`, and `IConfigurationProvider` for config settings.

Ping @timbussmann and @danielmarbach

### CODE REVIEW

File: nservicebus/logging/index_level_core_[6.0,).partial.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * Warn
+ * Error
+ * Fatal
+
+
+### Changing the Logging Level
+
+Note: When logging level is defined in both app.config and code the code wins.
+
+Warning: Configuration of logging via `app.config` is not recommended in Version 6, use the [code configuration API](#logging-levels-changing-the-logging-level-via-config-api) instead. The configuration section will be removed in Version 7.
```
Comment (timbussmann) [inline nservicebus/logging/index_level_core_[6.0,).partial.md]: it seems that logging displays the warning only for v6 while the other pages show the warning for all versions?

Comment (WilliamBZA) [inline nservicebus/logging/index_level_core_[6.0,).partial.md]: Logging and encryption are the only ones that had a partial that needed changing. The rest show for all the versions. This and encryption only appear if the user selects V6 from the version dropdown.

File: nservicebus/operations/auditing.md [hunk 2]
```diff
@@ -77,6 +77,8 @@ snippet:AuditWithCode
 
 ### Using app.config
 
+Warning: Configuration of auditing via `app.config` is not recommended in Version 6, use the [code configuration API](#configuring-auditing-using-code) instead. The configuration section will be removed in Version 7.
```
Comment (SimonCropp) [inline nservicebus/operations/auditing.md]: i know currently the warnings are specific to each use case. but can we come up with a generic one and put it in a `.include.md`?

eg

> Warning: Configuration this feature via `app.config` is not recommended in Version 6,  instead use the code based API listed above. The configuration section will be removed in Version 7.

Comment (WilliamBZA) [inline nservicebus/operations/auditing.md]: Done @SimonCropp. We lose the links to the code section, but I think that's fine. These warnings are all placed close enough to the code configuration sample to be on the same screen.

**Comment by SimonCropp** [comment]
@WilliamBZA LGTM. if u r done i say :shipit:

**Pull request merged**